Extensions/shorten_posts.js

```diff
@@ -17,163 +17,110 @@ XKit.extensions.shorten_posts = new Object({
 	height_default: 350,
 
 	preferences: {
-		sep0: {
-			text: "When to shorten posts",
-			type: "separator"
-		},
-		text_too: {
-			text: "Check and shorten text posts",
-			default: true,
-			value: true
-		},
-		photos_too: {
-			text: "Check and shorten photo posts and photosets",
-			default: false,
-			value: false
-		},
-		audio_too: {
-			text: "Check and shorten audio posts",
-			default: false,
-			value: false
-		},
-		asks_too: {
-			text: "Check and shorten replies to asks",
-			default: false,
-			value: false
-		},
-		videos_too: {
-			text: "Check and shorten video posts",
-			default: false,
-			value: false
-		},
-		chat_too: {
-			text: "Check and shorten chat posts",
-			default: false,
-			value: false
-		},
-		links_too: {
-			text: "Check and shorten link posts",
-			default: false,
-			value: false
-		},
-		quotes_too: {
-			text: "Check and shorten quote posts",
-			default: false,
-			value: false
-		},
 		height: {
 			text: "Maximum post height (<a id=\"xkit-shorten-posts-height-help\" href=\"#\" onclick=\"return false\">what is this?</a>)",
 			type: "text",
 			default: "350",
 			value: "350"
 		},
-		sep1: {
-			text: "Appearance and behaviour options",
-			type: "separator"
+		threshold: {
+			text: 'Minimum hidden height (<a id="xkit-shorten-posts-threshold-help" href="#" onclick="return false;">what is this?</a>)',
+			type: "text",
+			default: "50",
+			value: "50"
 		},
 		display_tags: {
 			text: "Display the tags on shortened posts",
 			default: true,
 			value: true
-		},
-		embiggen_on_click: {
-			text: "Embiggen photo posts when I click on the thumbnail",
-			default: true,
-			value: true
 		}
 	},
 
+	tagsCss: null,
+	wrapperCss: null,
+	avatarCss: null,
+	processing: false,
+
 	run: function() {
 
 		this.running = true;
 		XKit.extensions.shorten_posts.cpanel_check_height();
 
-		if ($(".posts .post").length > 0) {
+		if ($("[data-id], .posts .post").length > 0) {
```

Comment:
this check should be removed entirely in case the user hard-loads onto a page without posts and then soft-loads onto a page with posts